backend/cmd/headlamp.go

```diff
@@ -761,75 +761,62 @@ func isTokenAboutToExpire(token string) bool {
 	return time.Until(expiryTime) <= JWTExpirationTTL
 }
 
-//nolint:funlen
-func refreshAndCacheNewToken(oidcAuthConfig *kubeconfig.OidcConfig,
-	cache cache.Cache[interface{}], token string,
-) (string, error) {
-	const ExtendRefreshTokenTTL = 10 // seconds
-
-	// get provider
-	provider, err := oidc.NewProvider(context.Background(), oidcAuthConfig.IdpIssuerURL)
+func refreshAndCacheNewToken(clientID string, clientSecret string, refreshToken string, issuerURL string) (*oauth2.Token, error) {
+	// Get OIDC provider token URL
+	provider, err := oidc.NewProvider(context.Background(), issuerURL)
 	if err != nil {
-		logger.Log(logger.LevelError, nil, err, "failed to get provider")
-
-		return "", err
+		return nil, fmt.Errorf("error getting provider: %v", err)
 	}
 
-	// get refresh token from cache
-	refreshToken, err := cache.Get(context.Background(), fmt.Sprintf("oidc-token-%s", token))
-	if err != nil || refreshToken == "" {
-		logger.Log(logger.LevelError, nil, err, "failed to get refresh token")
-
-		return "", err
+	token, err := getNewTokenFromRefresh(clientID, clientSecret, refreshToken, provider.Endpoint().TokenURL)
+	if err != nil {
+		return nil, fmt.Errorf("error refreshing token: %v", err)
 	}
 
-	rToken, ok := refreshToken.(string)
-	if !ok {
-		logger.Log(logger.LevelError, nil, err, "failed to cast refresh token")
-
-		return "", err
+	if err := cacheRefreshedToken(token); err != nil {
+		return nil, fmt.Errorf("error caching refreshed token: %v", err)
 	}
 
-	oauth2Config := oauth2.Config{
-		ClientID:     oidcAuthConfig.ClientID,
-		ClientSecret: oidcAuthConfig.ClientSecret,
-		Endpoint:     provider.Endpoint(),
-		Scopes:       oidcAuthConfig.Scopes,
+	return token, nil
+}
+
+func getNewTokenFromRefresh(clientID string, clientSecret string, refreshToken string, tokenURL string) (*oauth2.Token, error) {
```

Comment:
Also please add comments for this function.

```suggestion
func getNewTokenFromRefresh(clientID, clientSecret, refreshToken, tokenURL string) (*oauth2.Token, error) {
```